src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -224,13 +224,20 @@ object LoadVCF {
       val hd = parseHeader(reader, getHeaderLines(hConf, file))
       val hd1 = header1Bc.value
 
-      hd1.sampleIds.iterator.zip(hd.sampleIds.iterator)
-        .zipWithIndex.dropWhile { case ((s1, s2), i) => s1 == s2 }.toArray.headOption match {
-        case Some(((s1, s2), i)) => fatal(
-          s"""invalid sample ids: expected sample ids to be identical for all inputs. Found different sample ids at position $i.
-           |    ${ files(0) }: $s1
-           |    $file: $s2""".stripMargin)
-        case None =>
+      if (hd1.sampleIds.length != hd.sampleIds.length) {
+        fatal(
+          s"""invalid sample ids: sample ids are different lengths.
+             | ${files(0)} has ${hd1.sampleIds.length} ids and
+             | ${file} has ${hd.sampleIds.length} ids.
+           """.stripMargin)
+      }
+
+      hd1.sampleIds.iterator.zipAll(hd.sampleIds.iterator, None, None)
+        .zipWithIndex.foreach { (i) =>
```

Comment:
This looks much better! you can replace `(i)` here with `case ((s1, s2), i) =>`. It will make all of the error message variables below easier to read.

Reply:
ahh cool I couldn't figure out how to do that. Let me fix

Reply:
fixed